### PR TITLE
[EISW-] [releases/2023/0] Rename VPU to NPU in openvino codebase

### PR DESCRIPTION
**Pull request by MirceaDan99**
### Details:
 - *item1*
 - *...*
 - PR for `master`: https://github.com/openvinotoolkit/openvino/pull/18969

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ inline const std::string get_plugin_lib_name_by_device(const std::string& device
             { "HETERO", "openvino_hetero_plugin" },
             { "BATCH", "openvino_auto_batch_plugin" },
             { "MULTI", "openvino_auto_plugin" },
-            { "VPUX", "openvino_intel_vpux_plugin" },
+            { "NPU", "openvino_intel_vpux_plugin" },
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:20]: openvino_intel_vpux_plugin => openvino_intel_npu_plugin ?

**Comment by MirceaDan99** [comment]
PR for `master`: https://github.com/openvinotoolkit/openvino/pull/18969

**Comment by ilya-lavrenov** [comment]
Please, port this change as well https://github.com/openvinotoolkit/openvino/pull/18204 for resolve issues with flakes8 on samples.

**Comment by ilya-lavrenov** [comment]
Two more PRs to fix Linux:
- https://github.com/openvinotoolkit/openvino/pull/18351
- https://github.com/openvinotoolkit/openvino/pull/18395

File: src/bindings/python/wheel/setup.py [hunk 2]
```diff
@@ -381,6 +382,16 @@ def generate_package(self, src_dirs):
...
 
+    user_options = [
+        ("skip-rpath", None, "Skips RPATH for Python extensions."),
+    ]
+    boolean_options = ["skip_rpath"]
+
+    def initialize_options(self):
+        """Set default values for all the options that this command supports."""
+        super().initialize_options()
+        self.skip_rpath = False
```
Comment (ilya-lavrenov) [inline src/bindings/python/wheel/setup.py:393]: I suppose changes in this file are not required

Comment (ilya-lavrenov) [reply src/bindings/python/wheel/setup.py]: I meant all changes

**Comment by wangleis** [comment]
Please refer https://github.com/openvinotoolkit/openvino/pull/18969 to update remained `vpu`.

File: src/tests/ie_test_utils/common_test_utils/test_constants.cpp [hunk 3]
```diff
@@ -11,7 +11,7 @@ const char *DEVICE_CPU = "CPU";
 const char *DEVICE_GNA = "GNA";
 const char *DEVICE_GPU = "GPU";
 const char *DEVICE_BATCH = "BATCH";
-const char *DEVICE_KEEMBAY = "VPUX";
+const char *DEVICE_KEEMBAY = "NPU";
```
Comment (PatrikStepan) [inline src/tests/ie_test_utils/common_test_utils/test_constants.cpp:14]: We should rename this to DEVICE_NPU and move it before BATCH

Comment (PatrikStepan) [reply src/tests/ie_test_utils/common_test_utils/test_constants.cpp:14]: There are few other occurrences of this in OV. Functional tests in vpux-plugin need to be updated as well.

File: tools/benchmark_tool/openvino/tools/benchmark/utils/constants.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Copyright (C) 2018-2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-VPUX_DEVICE_NAME = 'VPUX'
+VPUX_DEVICE_NAME = 'NPU'
```
Comment (PatrikStepan) [inline tools/benchmark_tool/openvino/tools/benchmark/utils/constants.py:4]: NPU_DEVICE_NAME?

**Comment by PatrikStepan** [comment]
Headers for src/inference/include/ie/vpu/ are still being used in some functional tests, so we can only remove that folder in another PR.
Other change requests were addressed by @StefaniaHergane  here: https://github.com/openvinotoolkit/openvino/pull/19002
Unfortunately, we will have to continue on that PR and close this one ( author will be OOO and not able to address reviews).

**Comment by PatrikStepan** [comment]
@ilya-lavrenov , can you please close this as a maintainer so we can continue on the other PR.

**Pull request closed without merging**